modules/rdf-tests/src/test/scala/com/gsk/rdf/tests/RdfTestsSpec.scala

```diff
@@ -51,14 +51,70 @@ class RdfTestsSpec extends AnyWordSpec with Matchers {
       .unsafeRunSync
   }
 
+  spark.stop()
+
   def assertTest(
       name: String,
       expected: List[Row],
       actual: List[Row]
   ): IO[Unit] = IO {
-    name in { expected.toSet shouldEqual actual.toSet }
+    name in {
+      val diff = expected.zip(actual).map((compareRows _).tupled).flatten
+      diff shouldBe Nil
+    }
+  }
+
+  def compareRows(expected: Row, actual: Row): Option[RowDiff] = {
+    if (expected.schema.fields.length != actual.schema.fields.length) {
+      Some(RowDiff(expected, actual))
+    } else {
+      expected.schema.fieldNames.foldLeft(Option.empty[RowDiff]) {
+        (diff, field) =>
+          val idx = expected.schema.fieldIndex(field)
+          if (!compareFields(expected.get(idx), actual.get(idx))) {
+            diff.orElse(Some(RowDiff(expected, actual)))
+          } else { diff }
```

Comment:
The handling of diffs seems a bit complex. This could be simplified to:
```suggestion
      val diff = expected.zip(actual).flatMap { case(expRow, actRow) =>
        if(!compareRows(expRow, actRow)) {
          Some(RowDiff(expRow, actRow))
        } else { None }
      }
      diff shouldBe Nil
    }
  }

  def compareRows(expected: Row, actual: Row): Boolean = {
    if (expected.schema.length != actual.schema.length) {
      false
    } else {
      expected.schema.fieldNames.forall { field =>
        compareFields(field, actual.get(expected.schema.fieldIndex(field)))
```